### PR TITLE
Issue 4

### LINKED ISSUE #4: Add Tests for all UVL Language Levels
Follow the guide on testing Lezer Parsers: https://lezer.codemirror.net/examples/test/

See the UVL examples referencend here: https://github.com/Universal-Variability-Language/java-fm-metamodel/blob/refactoring_metamodel/src/test/java/de/vill/parsing/ParsingTests.java and especially the files in https://github.com/Universal-Variability-Language/java-fm-metamodel/tree/refactoring_metamodel/src/test/resources/parsing

### PR DESCRIPTION
**Pull request by Ateyu**
Feat #4 

Adding tests for UVL parser.

### CODE REVIEW

File: src/grammar.grammar [hunk 1]
```diff
@@ -47,13 +47,13 @@ State{
...
 ConstraintSign { "|" | "=>" | "&" | "==" | "<=>" | "!="}
-ConstraintsItem { Neg? ((identifier | string | id) | ImportFeature) | BracketItem }
+ConstraintsItem { Neg? ((identifier | string | id | extIdentifier) | ImportFeature) | BracketItem }
 BracketItem { OpenBracket ConstraintsItem CloseBracket }
 Neg { "!" }
 Brackets { !first OpenBracket Constraints CloseBracket }
 OpenBracket { "(" }
 CloseBracket { ")" }
-Number { number | (OpenBracket Number NumericOperator Number CloseBracket)}
+Number { ("-"? number) | (OpenBracket Number NumericOperator Number CloseBracket)}
```
Comment (h3ssto) [inline src/grammar.grammar]: Switch to 
- UnsignedInteger for [0-9]+
- Integer for ([-]?)UnsignedInteger

File: src/grammar.grammar [hunk 2]
```diff
@@ -47,13 +47,13 @@ State{
 ConstraintsSection { "constraints" lineEnd ConstraintsBlock+ }
 
 ConstraintSign { "|" | "=>" | "&" | "==" | "<=>" | "!="}
-ConstraintsItem { Neg? ((identifier | string | id) | ImportFeature) | BracketItem }
+ConstraintsItem { Neg? ((identifier | string | id | extIdentifier) | ImportFeature) | BracketItem }
```
Comment (h3ssto) [inline src/grammar.grammar]: BooleanNeg

File: src/grammar.grammar [hunk 3]
```diff
@@ -105,13 +105,14 @@ lineEnd { newline | eof }
 }
 //!tokens
 @tokens {
-  @precedence {"mandatory", "optional", "alternative", "or", "constraints", "cardinality ","sum", "avg", "Integer", "Real","String", "Boolean","len", "-", "false", "true",real, number, identifier}
+  @precedence {"mandatory", "optional", "alternative", "or", "constraints", "cardinality ","sum", "avg", "Integer", "Real","String", "Boolean","len", "-", "false", "true",real, number, identifier, extIdentifier, id}
```
Comment (h3ssto) [inline src/grammar.grammar]: - fix comma ("true",real)
- What is the reason for this precedence order?

File: src/parser.mjs [hunk 4]
```diff
@@ -3,16 +3,16 @@ import {LRParser} from "@lezer/lr"
...
...fO#[!gO#i!jO#j!lO#k!lO~O!UzO!Tji~O!e!sO!f!sO~O!e!tO!i!tO~O!d!RO![Ua!]Ua!jUa!fUa!wUa#]Ua#^Ua#_Ua#bUa#cUa#dUa#eUa#fUa#gUa#hUa#WUa#`Ua#aUa~O!e!wO!i!wO~O!d!RO![Ua!]Ua!eUa!iUa!oUa!tUa~O!p!yO~O!e!zO!i!zO!v!{O~O!t![O![[a!][a~O!o!ZO!t![O![[a!][a~O!e!YO!i!YO!o!ZO!t![O![[a!][a~O!U#SO!V}a!]}a!a}a!e}a!i}a!l}a!m}a!n}a~O![tX!]tX!dVX!ftX!wtX#]tX#^tX#_tX#btX#ctX#dtX#etX#ftX#gtX#htX#WtX#`tX#atX~O#V#UO~O#V#VO~O#V#WO~O!e#YO!i#YO#V!iO#j#ZO#k#ZO~O#]#]O#^#]O#_#]O#`#]O#a#]O~O!f#_O!w#_O#]#aO#^#aO#_#`O#b#_O#c#_O#d#`O#e#`O#f#`O#g#`O#h#`O~O![YO!]YO~P-rO!V#hO!]#hO~P&bO!q#kO~O!e#mO~O!e#oO!i#oO!q#oO!t![O!v#qO!w#pO!y#nO!z#nO!{#oO!|cX!}cX~O!|#sO!}!uX~O!}#uO~O!t![O![[i!][i~O!o!ZO!t![O![[i!][i~O!p#xO#O#yO#P#yO#Q#yO#R#yO~O#W$PO~P-rO![ta!]ta!dVX!fta!wta#]ta#^ta#_ta#bta#cta#dta#eta#fta#gta#hta#Wta#`ta#ata~O!q$SO!w$TO#V!iO~O!q$SO!w$TO~P&bO!e!dO!i!dO!q$SO!w$TO#V!iO#i!jO#j!lO#k!lO~O!r$[O~O!v$]O~O!q$^O!{$^O~O!e$_O~O!|#sO!}!ua~O!t![O![[q!][q~O!V$gO!]$gO~P0uO#W$hO~O#W$iO~O#W$jO~O!q$kO~O!e#YO!i#YO#j#ZO#k#ZO~O#`#]O#a#]O![li!]li!fli!wli#]li#^li#_li#bli#cli#dli#eli#fli#gli#hli#Wli~O!q$oO~O!e$_O!v$rO~O!r$sO~O!f#_O!w#_O#b#_O#c#_O~O!e!dO!i!dO#V!iO#i!jO#j!lO#k!lO~O#W$PO~O!s$xO~O!s${O~O#O#P#Q#R#S!o#U#Y!l!n!m!a#[!w!z!y!{!q!e#k#j!m~",
```
Comment (h3ssto) [inline src/parser.mjs]: I guess this is from the parser generator? Isn't this generated by `@lezer/generator`? Why is it committed?

File: src/parser.mjs [hunk 5]
```diff
@@ -3,16 +3,16 @@ import {LRParser} from "@lezer/lr"
...
...*e#T#U!8Y#U#o*e~!8_V!eU}!O*y!Q![*e!c!}*e#R#S*e#T#b*e#b#c!8t#c#o*e~!8yV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#W*e#W#X!9`#X#o*e~!9eU!eU}!O*y!Q![*e!c!}*e#R#S*e#T#U!9w#U#o*e~!9|V!eU}!O*y!Q![*e!c!}*e#R#S*e#T#h*e#h#i!:c#i#o*e~!:hV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#c*e#c#d!:}#d#o*e~!;SV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#f*e#f#g!;i#g#o*e~!;nV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#m*e#m#n!<T#n#o*e~!<[T#O~!eU}!O*y!Q![*e!c!}*e#R#S*e#T#o*e~!<pX!eU}!O*y!Q![*e!c!}*e#R#S*e#T#d*e#d#e!=]#e#f*e#f#g!AS#g#o*e~!=bV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#h*e#h#i!=w#i#o*e~!=|V!eU}!O*y!Q![*e!c!}*e#R#S*e#T#]*e#]#^!>c#^#o*e~!>hV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#c*e#c#d!>}#d#o*e~!?SV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#b*e#b#c!?i#c#o*e~!?nU!eU}!O*y!Q![*e!c!}*e#R#S*e#T#U!@Q#U#o*e~!@VV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#`*e#`#a!@l#a#o*e~!@sT#P~!eU}!O*y!Q![*e!c!}*e#R#S*e#T#o*e~!AZT#R~!eU}!O*y!Q![*e!c!}*e#R#S*e#T#o*e~!AoV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#i*e#i#j!BU#j#o*e~!BZV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#a*e#a#b!Bp#b#o*e~!BwT#U~!eU}!O*y!Q![*e!c!}*e#R#S*e#T#o*e~!C]V!eU}!O*y!Q![*e!c!}*e#R#S*e#T#f*e#f#g!Cr#g#o*e~!CwV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#i*e#i#j!D^#j#o*e~!DcV!eU}!O*y!Q![*e!c!}*e#R#S*e#T#X*e#X#Y!Dx#Y#o*e~!EPT!y~!eU}!O*y!Q![*e!c!}*e#R#S*e#T#o*e~!EeO!t~~!EjO#d~~!EoO!}~~!EtO!]~",
```
Comment (h3ssto) [inline src/parser.mjs]: ...

**Review by h3ssto — CHANGES_REQUESTED** [review]

**Comment by h3ssto** [comment]
Currently, four tests fail / are false:
* `faulty/missingreference.uvl` is accepted by the Parser
* `concepts/namespace.uvl` is not accepted by the Parser
* `faulty/illegalname.uvl` Test behaviour is wrong, this file is expected to be invalid and therefore the test should not fail
* `faulty/wrongindent.uvl` Test behaviour is wrong, this file is expected to be invalid and therefore the test should not fail

File: src/grammar.grammar (hunk 1 above)
Comment (Ateyu) [reply src/grammar.grammar]: changed

File: src/grammar.grammar (hunk 3 above)
Comment (Ateyu) [reply src/grammar.grammar]: Thats a ranking if there are conflicts

Comment (h3ssto) [reply src/grammar.grammar]: Yes, that is the meaning of precedence but HOW can there be such conflicts? These words may no be used as feature or property names, always appear in a set manner. So this does not make sense to me.

**Pull request merged**